homeassistant/components/recorder/models.py

```diff
@@ -0,0 +1,115 @@
+"""Models for SQLAlchemy."""
+
+import json
+from datetime import datetime
+
+from sqlalchemy import (Boolean, Column, DateTime, ForeignKey, Index, Integer,
+                        String, Text, distinct)
+from sqlalchemy.ext.declarative import declarative_base
+import homeassistant.util.dt as dt_util
+from homeassistant.remote import JSONEncoder
+
+# SQLAlchemy Schema
+# pylint: disable=invalid-name
+Base = declarative_base()
+
+
+class Events(Base):
+    # pylint: disable=too-few-public-methods
+    """Event history data."""
+
+    __tablename__ = 'events'
+    event_id = Column(Integer, primary_key=True)
+    event_type = Column(String(32), index=True)
+    event_data = Column(Text)
```

Comment:
When I last looked, the JSON type was supported only with dialects that understood JSON and that there was no fallback yet for other databases.  
http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html#json-types

I didn't want to get into using the complexity of custom type converter at this time, as it's likely before long SQLAlchemy core will support a gracefully degrading JSON type.

However, it's been over 6 weeks since I last looked at SQLAlchemy's types and no progress, so perhaps it's worth a custom type converter to further simplify the code.